src/types.ts

```diff
@@ -1,16 +1,42 @@
 /**
  * A universal API to make network requests. A subset of the `fetch()` API.
+ *
+ * {@link https://developer.mozilla.org/en-US/docs/Web/API/fetch}
  */
 export type FetchLike = (
 	input: string,
 	init?: RequestInitLike,
 ) => Promise<ResponseLike>;
 
+/**
+ * An object that allows you to abort a `fetch()` request if needed via an
+ * `AbortController` object
+ *
+ * {@link https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal}
+ */
+// `any` is used often here to ensure this type is universally valid amond
+// different AbortSignal implementations. The types of each property are not
+// important to validate since it is blindly passed to a given `fetch()`
+// function.
+export type AbortSignalLike = {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	aborted: any;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	addEventListener: any;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	removeEventListener: any;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	dispatchEvent: any;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	onabort: any;
+};
+
 /**
  * The minimum required properties from RequestInit.
  */
 export interface RequestInitLike {
 	headers?: Record<string, string>;
+	signal?: AbortSignalLike | null;
```

Comment:
Is it ever null?

Reply:
I'm not sure if realistically it is ever `null`, but TypeScript's global type for `fetch()` includes `null`. Without it being added here, you get a type error.

https://github.com/microsoft/TypeScript/blob/3328feb7991f358e245088d48b64ad9da8f015e2/lib/lib.dom.d.ts#L1515-L1518

Reply:
I added a comment so it's clear why it's added. It does look pretty odd to include.